File: backend/streamcraft/domain/vod/value_objects/platform.py
"""Platform value object."""

from enum import Enum


class Platform(str, Enum):
    """Streaming platform."""

    TWITCH = "twitch"
    YOUTUBE = "youtube"

    def __str__(self) -> str:
        """String representation."""
        return self.value

    @property
    def display_name(self) -> str:
        """Get human-readable display name."""
        return self.value.title()

File: frontend/src/domain/vod/value-objects/platform.ts
/**
 * Platform value object.
 */

export enum Platform {
    Twitch = 'twitch',
    YouTube = 'youtube',
}

export const getDisplayName = (platform: Platform): string => {
    switch (platform) {
        case Platform.Twitch:
            return 'Twitch';
        case Platform.YouTube:
            return 'YouTube';
    }
};

export const fromString = (value: string): Platform | null => {
    const normalized = value.toLowerCase();
    if (normalized === 'twitch') return Platform.Twitch;
    if (normalized === 'youtube') return Platform.YouTube;
    return null;
};
